java-compiler-testing/src/main/java/io/github/ascopes/jct/containers/impl/ModuleContainerGroupImpl.java

```diff
@@ -93,19 +96,19 @@ public void addModule(String module, PathWrapper path) {
 
   @Override
   public void close() throws IOException {
-    var exceptions = new ArrayList<IOException>();
+    var exceptions = new ArrayList<Throwable>();
     for (var group : modules.values()) {
       try {
         group.close();
-      } catch (IOException ex) {
+      } catch (Exception ex) {
         exceptions.add(ex);
       }
     }
 
     if (!exceptions.isEmpty()) {
-      var ex = new IOException("One or more module impl failed to close");
-      exceptions.forEach(ex::addSuppressed);
-      throw ex;
+      var newEx = new IOException("One or more module impl failed to close");
+      exceptions.forEach(newEx::addSuppressed);
```

Comment:
- [ ] Maybe this could go in IoExceptionUtils. If we use this logic elsewhere for non-IO exceptions, I could probably repurpose IoExceptionUtils as just ExceptionUtils

Reply:
Not going to do this for now, can see little benefit.